MapboxCoreNavigation/RouteStep.swift

```diff
@@ -38,4 +39,46 @@ extension RouteStep {
     open var lastInstruction: SpokenInstruction? {
         return instructionsSpokenAlongStep?.last
     }
+    
+    /**
+     Returns true if the current route step contains a tunnel.
+     */
+    var containsTunnel: Bool {
+        guard let intersections = intersections else { return false }
+        for intersection in intersections {
+            if intersection.outletRoadClasses?.contains(.tunnel) == true {
+                return true
+            }
+        }
+        return false
+    }
+    
+    /**
+     Returns a tunnel slice for the current route step coordinates
+     */
+    var tunnelSlice: Polyline? {
+        guard let coordinates = coordinates, let intersectionBounds = tunnelIntersectionBounds else { return nil }
+        return Polyline(coordinates).sliced(from: intersectionBounds.entry.location, to: intersectionBounds.exit.location)
+    }
+    
+    /**
+     Returns the distance of a tunnel slice for the current route step
+     */
+    var tunnelDistance: CLLocationDistance? {
+        guard let tunnelSlice = tunnelSlice else { return nil }
+        return tunnelSlice.distance()
+    }
+    
+    /**
+     Returns the entry and exit intersection bounds of the tunnel on the current route step
+     */
+    var tunnelIntersectionBounds: (entry: Intersection, exit: Intersection)? {
+        guard let intersections = intersections, containsTunnel else { return nil }
+        for i in 0..<(intersections.count - 1) where intersections.count > 1 {
```

Comment:
A couple of things I notice here. 

* I'm not sure `where intersections.count > 1 ` is necessary to check. Looping over a count of 1 is fine.
* The above point does not guard against `i + 1`. We should do that check while in the loop.
* it might be cleaner to do `for (intersectionIndex, intersection)` in intersections.enumerated() {...`